stan/math/prim/err/check_nonzero_size.hpp

```diff
@@ -21,11 +21,13 @@ namespace math {
 template <typename T_y>
 inline void check_nonzero_size(const char* function, const char* name,
                                const T_y& y) {
-  if (y.size() > 0) {
-    return;
+  STAN_NO_RANGE_AND_SIZE_CHECK
+  if (y.size() == 0) {
+    [&]() STAN_COLD_PATH {
+      invalid_argument(function, name, 0, "has size ",
+                       ", but must have a non-zero size");
```

Comment:
I am not sure if wrapping a single function call into a lambda affects performance. Hmm, maybe it does, as the arguments have to be passed into `invalid_argument`. Anyway You could also add `STAN_COLD_PATH` attribute to `invalid_argument`.

Reply:
The invalid argument code is just 

```
  std::ostringstream message;
  message << function << ": " << name << " " << msg1 << y << msg2;
  throw std::invalid_argument(message.str());
```

But idt we want the computer to even worry about having to construct that message if the check is going to pass